src/protocols/rdp/plugins/guacai/guacai-messages.c

```diff
@@ -48,8 +52,9 @@ static void guac_rdp_ai_read_format(wStream* stream,
     Stream_Read_UINT16(stream, format->bps); /* wBitsPerSample */
     Stream_Read_UINT16(stream, format->data_size); /* cbSize */
 
-    /* Read arbitrary data block (if applicable) */
-    if (format->data_size != 0) {
+    /* Read arbitrary data block (if applicable) and data is available. */
+    if (format->data_size != 0
+            && Stream_GetRemainingLength(stream) >= format->data_size) {
```

Comment:
This may produce some odd behavior. The check is necessary, but the failure status may need to be carried outward to the caller of `guac_rdp_ai_read_format()`.

This function is invoked by `guac_rdp_ai_process_formats()` in a loop, using each call to process one of the received audio input formats. If this check fails for one of the formats, and other formats remain to be read, then their headers will start to be read within what theoretically is the data portion of the preceding format.

Reply:
Okay, so maybe change this from a `static void` to `static int` and return something non-zero if this happens?

Reply:
Okay, took a stab at changing this such that any errors in `guac_rdp_ai_read_format()` results in `guac_rdp_ai_process_formats()` bailing out without continuing trying to process formats.  Let me know if that needs any work.

Reply:
Well, my initial try didn't work so well - it just bails out and doesn't process any sound input :-).  Apparently I need to try something else...

Reply:
Okay, I think I got it - have to deal with the availability of the arbitrary data block separately from checking the Stream for the amount of data.  Once I split that up it works as expected.